### PR TITLE
[fix] Unused arguments in lib_stusb_impl/usbd_impl.c:USBD_HID_DataOut_impl when HAVE_USB_HIDKBD macro is set

### PR DESCRIPTION
**Pull request by lpascal-ledger**
## Description

Please provide a detailed description of what was done in this PR.
(And mention any links to an issue [docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

## Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

## Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: lib_stusb_impl/usbd_impl.c [hunk 1]
```diff
@@ -1018,8 +1018,13 @@ uint8_t USBD_HID_DataIn_impl(USBD_HandleTypeDef *pdev, uint8_t epnum)
...
                               uint8_t             epnum,
-                              uint8_t            *buffer,
-                              apdu_buffer_t      *apdu_buf)
+#ifndef HAVE_USB_HIDKBD
+                              uint8_t       *buffer,
+                              apdu_buffer_t *apdu_buf)
+#else
+                              __attribute__((unused)) uint8_t       *buffer,
+                              __attribute__((unused)) apdu_buffer_t *apdu_buf)
+#endif  // HAVE_USB_HIDKBD
```
Comment (xchapron-ledger) [inline lib_stusb_impl/usbd_impl.c:1027]: Maybe this to confuse less IDE?
```suggestion
#ifndef HAVE_USB_HIDKBD
                              uint8_t       *buffer,
                              apdu_buffer_t *apdu_buf
#else
                              __attribute__((unused)) uint8_t       *buffer,
                              __attribute__((unused)) apdu_buffer_t *apdu_buf
#endif  // HAVE_USB_HIDKBD
                             )
```

**Review by xchapron-ledger — COMMENTED** [review]
Can you rename the commit / Pr to explicit what part of the code it impact?

File: lib_stusb_impl/usbd_impl.c (hunk 1 above)
Comment (lpascal-ledger) [reply lib_stusb_impl/usbd_impl.c:1027]: I'll update, but I bet the linter will whine about something.

**Review by xchapron-ledger — APPROVED** [review]

**Comment by lpascal-ledger** [comment]
`codecov-action` decided it could not be bothered to uploaded the report anymore :shrug:

**Comment by xchapron-ledger** [comment]
> `codecov-action` decided it could not be bothered to upload the report anymore 🤷

That's sadly not unusual.

**Pull request merged**